.github/workflows/maven.yml

```diff
@@ -18,14 +18,72 @@ jobs:
   build:
 
     runs-on: ubuntu-latest
-
     steps:
     - uses: actions/checkout@v3
     - name: Set up JDK 17
       uses: actions/setup-java@v3
       with:
         java-version: '17'
         distribution: 'temurin'
-        cache: maven
+    - name: Cache local Maven repository
+      uses: actions/cache@v3
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('pom.xml', '*/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
```

Comment:
Is there a reason for this change? The `cache: maven` attribute was created to avoid this.

Reply:
I use the `cache: maven` but it did not work as expected. Somehow the generated `pom.xml` were changing the cache key and the maven artefacts were always re-downloaded. This change is to take into account `pom.xml` and `*/pom.xml` instead of `**/pom.xml`.
I can try it again though to make sure there were no other changes that was impacting the Maven cache

Reply:
Ah, okay. I was just curious. TBH I've seen it be hit or miss on what cache it uses, but I've not noticed too many issues with it. However, I've got no strong opinion either way.

Reply:
@jamezp thanks for the comment. I fixed it and reverted to use `cache: maven`. At the end of the CI, I clean up any generated files so that their `pom.xml` are not hashed.